### PR TITLE
Swap widget updates

### PR DESCRIPTION
**Pull request by bitsforpeace**
### What does this PR do?

Use WidgetButton
Use WidgetContainer
start adding button texts
Add transaction screens

TODO:
buttons on tx screens
etherscan links

### CODE REVIEW

File: packages/widgets/src/SwapWidget/index.tsx [hunk 1]
```diff
@@ -113,62 +121,93 @@ function SwapWidgetWrapped({
...
-        </Box>
-      </CardBody>
-    </Card>
+      }
+      footer={address ? <WidgetButton amount={amount} onClickAction={execute} /> : connectButton}
+    >
+      {txStatus !== TxStatus.IDLE ? (
+        <SwapTransactionStatus />
+      ) : (
+        <VStack align="stretch">
```
Comment (codewiththeflowxyz) [inline packages/widgets/src/SwapWidget/index.tsx:176]: I think we can extract this part to a separate component <Swap>

**Review by codewiththeflowxyz — APPROVED** [review]

File: packages/widgets/src/SwapWidget/index.tsx [hunk 2]
```diff
@@ -113,62 +130,138 @@ function SwapWidgetWrapped({
...
-              <Button onClick={execute}>{isQuoteLoading ? 'Loading...' : 'Swap'}</Button>
-            ) : (
-              connectButton
-            )}
-          </VStack>
-        </Box>
-      </CardBody>
-    </Card>
+      }
+      footer={address ? <WidgetButton amount={amount} onClickAction={onClickAction} /> : connectButton}
```
Comment (generatorpoint) [inline packages/widgets/src/SwapWidget/index.tsx:225]: There's a couple of "breaking changes" compared to the PR I opened for passing a connect function into the widget. eg the `WidgetButton` doesn't need to take the amount, [see this](https://github.com/jetstreamgg/helipad/pull/40/files#diff-7870f7fb9b5d492f25cf68b824d43505ba6c2c4a83b13c0f179423a1e1052f59L388-R414)

the reason the WidgetButton had the amount was to update some text, but that logic can be done in the `setButtonText` block a little bit above. [See this](https://github.com/jetstreamgg/helipad/pull/40/files#diff-7870f7fb9b5d492f25cf68b824d43505ba6c2c4a83b13c0f179423a1e1052f59R380-R381)

File: packages/widgets/src/SwapWidget/index.tsx [hunk 3]
```diff
@@ -113,62 +130,138 @@ function SwapWidgetWrapped({
...
   });
 
+  useEffect(() => {
+    //Initialize the withdraw flow
+    setWidgetState({
+      flow: SwapFlow.SWAP,
+      action: SwapAction.SWAP,
+      screen: SwapScreen.ACTION
+    });
+  }, []);
```
Comment (generatorpoint) [inline packages/widgets/src/SwapWidget/index.tsx:140]: Also the widget state is now initialized as null so we can handle when the account is disconnected, eg setting the data-testid [like here](https://github.com/jetstreamgg/helipad/pull/40/files#diff-c3a53099d5b1f88d98c402a0bae584d59b265e04bd9a4cee8968c098f8064bc6R180-R187).

Which means we have to reset the widget state after disconnecting [like this](https://github.com/jetstreamgg/helipad/pull/40/files#diff-c3a53099d5b1f88d98c402a0bae584d59b265e04bd9a4cee8968c098f8064bc6R197-R222)

**Comment by bitsforpeace** [comment]
Going to merge this into the other branch so all the swap stuff is all together before we start fixing conflicts with other branches

**Pull request merged**